### PR TITLE
MYEN-409 Improvement of React client example

### PR DESCRIPTION
**Pull request by jrhender**
Modified the React example to be essentially the same as the current Vue example, but with React colours.
(Great job on the Vue example @dwojno , made this pretty easy 👍 )

### CODE REVIEW

File: examples/react-dapp/src/App.tsx [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import React, { useState } from 'react'
 import './App.css'
+import './Login.css'
 import { IAM, CacheServerClient } from 'iam-client-lib'
 import axios from 'axios'
-import Button from '@material-ui/core/Button'
-import { Typography } from '@material-ui/core'
```
Comment (dwojno) [inline examples/react-dapp/src/App.tsx:6]: If you're not using material ui, please remove it from package.json 😄

File: examples/react-dapp/src/App.tsx [hunk 2]
```diff
@@ -18,52 +21,129 @@ type Role = {
...
   }
-  return (
-    <div className="App">
-      <header className="App-header">
+
+  const logout = function () {
+    setDID('');
+  }
+
+  let loginJsx
```
Comment (dwojno) [inline examples/react-dapp/src/App.tsx:85]: I would assign each view to different function like renderLogin, renderLoading, renderRoles, and main function with
`
if () {
return renderLogin()
}
if () {
return renderButton()
}
return renderRoles()
`
and run this function inside jsx

File: examples/react-dapp/src/App.tsx (hunk 1 above)
Comment (jrhender) [reply examples/react-dapp/src/App.tsx:6]: Nice catch, thanks, I've removed it

File: examples/react-dapp/src/App.tsx [hunk 3]
```diff
@@ -18,52 +21,138 @@ type Role = {
...
+    if (isLoading) {
+      return renderLoading()
+    }
+    else if (did) {
+      return renderLoginResults();
+    }
+    else {
+      return renderLoginOptions();
+    }
+  };
```
Comment (jrhender) [inline examples/react-dapp/src/App.tsx:148]: @dwojno Is this what you had in mind for the `loginJsx` refactor or is there a further improvement?

File: examples/react-dapp/src/App.tsx (hunk 2 above)
Comment (jrhender) [reply examples/react-dapp/src/App.tsx]: Good point, it's better to not be mutating the `loginJsx` variable

**Review by manihagh — APPROVED** [review]

File: examples/react-dapp/src/App.tsx (hunk 3 above)
Comment (dwojno) [reply examples/react-dapp/src/App.tsx:148]: Yes, perfect 👍

**Review by dwojno — APPROVED** [review]

**Pull request merged**